src/app/core/services/authorization.service.ts

```diff
@@ -5,27 +5,40 @@ import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { LoginForm, Token } from '../models/authorization';
 import { User } from '../models/user';
 import { catchError, Observable, tap, throwError } from 'rxjs';
+import { LoaderService } from '.';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthorizationService {
   private apiUrl = 'http://localhost:3004/auth';
   private isAuth: boolean = !!localStorage.getItem('accessToken');
+  public userData: Observable<User> = new Observable();
 
   private httpOptions = {
     headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
   };
 
-  constructor(private router: Router, private http: HttpClient) {}
+  constructor(
+    private router: Router,
+    private http: HttpClient,
+    private loaderService: LoaderService
+  ) {
+    if (this.isAuthenticated()) {
+      this.userData = this.getUserInfo();
+    }
+  }
 
   login(loginFormData: LoginForm): Observable<Token> {
+    this.loaderService.setLoading(true);
     return this.http
       .post<Token>(`${this.apiUrl}/login`, loginFormData, this.httpOptions)
       .pipe(
         tap((data: Token) => {
           localStorage.setItem('accessToken', data.token);
           this.isAuth = true;
+          this.userData = this.getUserInfo();
+          this.loaderService.setLoading(false);
         }),
```

Comment:
then you can do immediate chain operator for another API call:

`switchMap(() => this.getUserInfo())`

Reply:
done